scripts/servers.py

```diff
@@ -722,3 +722,47 @@ def managed_webdriverio_server(
             yield proc
     finally:
         del os.environ['MOBILE']
+
+
+@contextlib.contextmanager
+def managed_acceptance_tests_server(
+    suite_name: str,
+    stdout: int = subprocess.PIPE,
+) -> Iterator[psutil.Process]:
+    """Returns context manager to start/stop the Acceptance server gracefully.
+
+    Args:
+        suite_name: str. The suite name whose tests should be run.
+        stdout: int. This parameter specifies the executed program's standard
+            output file handle.
+
+    Yields:
+        psutil.Process. The jasmine testing process.
+    """
+    nodemodules_jasmine_bin_path = os.path.join(
+        common.NODE_MODULES_PATH, '.bin', 'jasmine')
+    puppeteer_acceptance_tests_dir_path = os.path.join(
+        common.CURR_DIR, 'core', 'tests', 'puppeteer-acceptance-tests')
+    spec_dir_path = os.path.join(
+        puppeteer_acceptance_tests_dir_path, 'spec')
+    jasmine_config_file_path = os.path.join(
+        puppeteer_acceptance_tests_dir_path, 'jasmine.json')
+
+    acceptance_tests_args = [
+        nodemodules_jasmine_bin_path,
+        '--config="%s"' % jasmine_config_file_path,
+        '%s' % os.path.join(spec_dir_path, suite_name)
+    ]
+
+    # OK to use shell=True here because we are passing string literals and
+    # constants, so there is no risk of a shell-injection attack.
+    managed_acceptance_tests_proc = managed_process(
+        acceptance_tests_args,
```

Comment:
This includes `suite_name`, which is user-controlled. Therefore, there is a risk of a shell-injection attack here (i.e. it is not true that we are only passing string literals and constants). To mitigate this, we need to validate the suite name (either that it's a valid suite name or that it only contains "safe" characters that cannot trigger a shell injection)

Reply:
done this. PTAL